### PR TITLE
BUGFIX: If python2js_buffer fails, python2js leaves the error flag in an invalid state

### LINKED ISSUE #1110: Soundness error in python2js_buffer
This bug goes back at least to version 0.15.0. It's been messing up my error handling code.
First:
```python
import numpy as np
x = np.array([['string1', 'string2'], ['string3', 'string4']])
```
Second: type `x<enter>1<enter>x<enter>1<enter>`.

### PR DESCRIPTION
**Pull request by hoodmane**
I fixed #1110. If `python2js_buffer` fails, we need to clear the Python error flag before trying a backup approach.

### CODE REVIEW

**Comment by dalcde** [comment]
Can you test this without using numpy? This ought to be a type conversion test.

**Comment by hoodmane** [comment]
I don't know, will investigate. I'm going to update `conftest` so that it checks whether `PyErr_Occurred()` at the end of every test too, that would have caught this bug.

**Comment by hoodmane** [comment]
I think we should also try converting the numpy tests from `selenium_standalone` to `selenium`. I am suspicious that the reason they were changed to `selenium_standalone` in the first place may have been because of this bug.

**Comment by dalcde** [comment]
Also, what's the difficulty in turning the original bug reproduction
method into an actual test case?

**Comment by hoodmane** [comment]
Original bug reproduction involved the code in `console.html` in a weird way.

**Comment by dalcde** [comment]
You will have to load the numpy package first.

**Comment by hoodmane** [comment]
Indeed I should.

File: src/tests/test_typeconversions.py [hunk 1]
```diff
@@ -327,3 +327,20 @@ def test_javascript_error_back_to_js(selenium):
...
+    selenium.run(
+        """
+        import array
+        a = array.array("Q", [1,2,3])
+        """
+    )
+    selenium.run_js(
+        """
+        pyodide.globals.a
+        if(pyodide._module._PyErr_Occurred()){
```
Comment (dalcde) [inline src/tests/test_typeconversions.py:342]: This check would be made redundant by #1113, right?

Comment (dalcde) [reply src/tests/test_typeconversions.py:342]: If so, I think the correct course of action is to merge this and then remove this line in #1113

Comment (hoodmane) [reply src/tests/test_typeconversions.py:342]: Either we should remove this here and keep the line in #1113 or keep the line here and close #1113.

Comment (dalcde) [reply src/tests/test_typeconversions.py:342]: I meant merge this test, and then in #1113 remove this particular line in this test

Comment (hoodmane) [reply src/tests/test_typeconversions.py:342]: Oh yeah. I agree.

**Pull request merged**